internal/controller/runtime/fsm/runtime_fsm_apply_clusterrolebindings.go

```diff
@@ -0,0 +1,192 @@
+package fsm
+
+import (
+	"context"
+	"slices"
+
+	authenticationv1alpha1 "github.com/gardener/gardener/pkg/apis/authentication/v1alpha1"
+	gardener_api "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	//nolint:gochecknoglobals
+	labelsClusterRoleBindings = map[string]string{
+		"app":                                   "kyma",
+		"reconciler.kyma-project.io/managed-by": "kim",
+	}
+)
+
+func sFnApplyClusterRoleBindings(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	// prepare subresource client to request admin kubeconfig
+	srscClient := m.ShootClient.SubResource("adminkubeconfig")
+	shootAdminClient, err := GetShootClient(ctx, srscClient, s.shoot)
+	if err != nil {
+		updateCRBApplyFailed(&s.instance)
+		return updateStatusAndStopWithError(err)
```

Comment:
Are we OK with returning error from reconcile function?

Reply:
It's the same as if we would log error and reque:
- https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler